### DISCUSSION TITLE
Custom Handlebars helper in custom layout

### DISCUSSION THREAD

**Discussion by sebnunes** [microsoft-search/pnp-modern-search#1759] [Q&A] [answered]
Hi folks, 

First thanks to the contributors of the PnP modern search :-) I love it! 

Quick question, I would like to add a custom handlebars helper to manipulate a string coming from a metadata in my result. 

I added my custom JS before the div for the template 👍:
`<script type="text/javascript">
    Handlebars.registerHelper('trimString', function(passedString) {
        var theString = passedString.substring(8);
    return new Handlebars.SafeString(theString)
    });
</script>
    <div class="template">
...` 

Then using the custom helper like this: 
`{{trimString (slot item @root.slots.Category)}}`

But I get an error "Uncaught (in promise) ".

Any clue how to troubleshoot this? 

Thanks. 

Sébastien

**Comment by wobba** [comment, marked as answer]
That approach doesn't work. You need to look at https://microsoft-search.github.io/pnp-modern-search/extensibility/handlebars_customizations/

**Reply by sebnunes** [reply]
Great thanks for your answer!
